ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -80,12 +80,39 @@ private[platform] class MutableCacheBackedContractStore(
       .flatMap {
         case (cached, toBeFetched) if toBeFetched.isEmpty =>
           Future.successful(Some(cached.max))
+
         case (cached, toBeFetched) =>
-          contractsReader
-            .lookupMaximumLedgerTime(toBeFetched)
-            .map(_.map(m => (cached + m).max))
+          readThroughMaximumLedgerTime(toBeFetched.toList, cached.maxOption)
       }
 
+  private def readThroughMaximumLedgerTime(
+      missing: List[ContractId],
+      acc: Option[Timestamp],
+  ): Future[Option[Timestamp]] =
+    missing match {
+      case contractId :: restOfMissing =>
+        readThroughContractsCache(contractId).flatMap {
+          case active: Active =>
+            val newMaximumLedgerTime = Some(
+              (active.createLedgerEffectiveTime :: acc.toList).max
+            )
+            readThroughMaximumLedgerTime(restOfMissing, newMaximumLedgerTime)
+
+          case _: Archived =>
+            Future.failed(MissingContracts(Set(contractId)))
```

Comment:
We are not reporting anymore all missing contract ids, but only the first one detected as missing. I'd update the logs handling this error to make sure no confusion can arise when checking them

https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutor.scala#L93-L94 

I don't have a strong opinion on the deterministic assertion

Reply:
This is somewhat fuzzy for me what is needed here, not sure we reported from the mutable cache all missing contracts ever, see:
* this implementation reports all archived contracts, which can be found in the cache, or if none, the first missing contract looked up with this logic. I think this is still consistent with the log message, stating `Some input contracts could not be found`
* the previous implementation was not reporting all neither: if in the cache there was already some missing, it was reporting those combined, without reporting or checking all contracts, which are not in the cache. It think this is also consistent with the log message though.
* earlier it was even more strange: If there were archived contracts in the cache, then we reported all contracts (?) if there was not, then we reported all contracts which are missing (the db query).
* I think originally we might have reported all contract which were missing (only the DB query)

Since we not really seemed to aim to report all missing contracts for a while, and since this is also consistent with the log message, I thought I just aim for the most efficient solution in this space, although happy to improve/change if there is a need.

Reply:
You are right that it was never fully consistent with the message. But let's make it right if you see reason. The message starts with `Some contracts missing..` which makes sense. But in the end, it says `..Missing contracts: [<the-missing-contract-ids>` which might hint at a comprehensive list being forwarded. I would re-phrase the log message as 
```
s"Some input contracts could not be found: ${contracts.mkString("[", ", ", "]")}. Restarting the computation."
`

Reply:
Make sense to me, asking @gerolf-da as well, since he refactored this the last time: wdyt?

Reply:
Let's move this discussion to the next PR where I will refactor the ContractStore.lookupMaximumLedgerTime